### PR TITLE
Add 7.1 compatibility

### PR DESCRIPTION
**Pull request by nenad**
# What this PR changes:
- Adds php 7.1 compatibility

### CODE REVIEW

**Comment by boekkooi-fresh** [comment]
:+1:  I still :cry: that you didn't up the required PHP version

[Approved with PullApprove](https://pullapprove.com/hellofresh/engine/pull-request/9/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=hellofresh/engine)

**Comment by emilyclaudet** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/hellofresh/engine/pull-request/9/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=hellofresh/engine)

File: tests/EventStore/EventStoreTest.php [hunk 1]
```diff
@@ -96,7 +95,7 @@ protected function createDomainMessage($id, $version, $recordedOn = null)
         return new DomainMessage(
             $id,
             $version,
-            new SomethingHappened(),
+            new SomethingHappened($recordedOn),
             $recordedOn ? $recordedOn : new \DateTimeImmutable("now")
```
Comment (sergiughf) [inline tests/EventStore/EventStoreTest.php]: Why not use the new php 7 null coalesce operator here? $recordedOn ?? new \DateTimeImmutable("now")

File: tests/Mock/SomethingHappened.php [hunk 2]
```diff
@@ -14,9 +14,9 @@ class SomethingHappened implements DomainEventInterface
     /**
      * SomethingHappened constructor.
      */
-    public function __construct()
+    public function __construct(\DateTimeInterface $dateTime = null)
     {
-        $this->occurredOn = new \DateTime();
+        $this->occurredOn = $dateTime === null ? $dateTime : new \DateTime();
```
Comment (sergiught) [inline tests/Mock/SomethingHappened.php:19]: Why not use the new php 7 null coalesce operator here as well? 

```
$dateTime ?? new \DateTime();
```

Comment (nenad) [reply tests/Mock/SomethingHappened.php:19]: This needs to be compatible with php 5.6

**Comment by sergiught** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/hellofresh/engine/pull-request/9/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=hellofresh/engine)

**Comment by sergiught** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/hellofresh/engine/pull-request/9/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=hellofresh/engine)

**Comment by boekkooi-fresh** [comment]
:+1: 

[Approved with PullApprove](https://pullapprove.com/hellofresh/engine/pull-request/9/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=hellofresh/engine)

**Pull request merged**